src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1144,7 +1154,11 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         // Don't add the tag and changelog if we've already processed this BuildData before.
         if (!buildDataAlreadyPresent) {
-            build.addAction(new GitTagAction(build, workspace, revToBuild.revision));
+            if (build.getActions(AbstractScmTagAction.class).isEmpty()) {
```

Comment:
`build.getAction(AbstractScmTagAction.class) == null` is more effective from the performance PoV. NIT